Show Python path for Poetry in `debug info`

jaredahern opened the pull request:
# Pull Request Check List

Resolves: #10559

- [x] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

# Summary

Adds the executable and environment path for the Python executable running Poetry to `poetry self debug`.

I believe the `sys` calls are correct, but happy to adjust as needed. Also, I did not see any docs that required updating.

## Summary by Sourcery

Include the Python environment path and executable in the output of `poetry self debug`.

Enhancements:
- Add 'Path' and 'Executable' fields under the Poetry section in debug info
- Adjust label alignment for Version and Python fields

Tests:
- Add tests to verify the display of Poetry's path and executable in debug info

radoering reviewed (commented):
LGTM except for the failing Windows tests. Can you take a look, please?

jaredahern commented:
Hmm. I took a look at the errors, and tried to resolve them. Not sure if I totally have. Here's the original error:

```
    Poetry
    Version:    2.2.1
    Python:     3.10.11
    Path:       /poetry/prefix
    Executable: /poetry/prefix/bin/python

    Virtualenv
    Python:         3.7.0
    Implementation: CPython
    Path:           \prefix
  - Executable:     /poetry/prefix/bin/python
  ?                 ^      ^      ^   ^
  + Executable:     \poetry\prefix\bin\python
  ?                 ^      ^      ^   ^
    Valid:          True

    Base
    Platform:   darwin
    OS:         posix
    Python:     3.10.11
    Path:       \base\prefix
    Executable: python
```

So here, the error appears to be with the output from the `env info` command, which I haven't touched, and directly copied the test for. It confuses me that the related `test_env_info_displays_complete_info` test does not fail.

That being said, I tried to be more careful with paths, to better support Windows. Happy to take pointers. I rebased onto main and added a commit. There are now different failing tests, which seem to be unrelated to the changes here. Any thoughts on how I should proceed?

radoering commented:
> So here, the error appears to be with the output from the `env info` command, which I haven't touched, and directly copied the test for. It confuses me that the related `test_env_info_displays_complete_info` test does not fail.

You mock `sys.executable` in the new test. Your mock uses forward slashes on Windows. `test_env_info_displays_complete_info` just uses the real `sys.executable`, which has backslashes.

> There are now different failing tests, which seem to be unrelated to the changes here. Any thoughts on how I should proceed?

Just wait and re-run. That was https://github.com/actions/runner-images/issues/13341
